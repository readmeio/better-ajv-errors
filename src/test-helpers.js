import { readFileSync } from 'fs';
import path from 'path';

export async function getSchemaAndData(name, dirPath) {
  const schemaPath = path.join(dirPath, '..', '__fixtures__', name, 'schema.json');
  const schema = JSON.parse(readFileSync(schemaPath, 'utf8'));
  const dataPath = path.join(dirPath, '..', '__fixtures__', name, 'data.json');
  const json = readFileSync(dataPath, 'utf8');
  const data = JSON.parse(json);

  return [schema, data, json];
}
