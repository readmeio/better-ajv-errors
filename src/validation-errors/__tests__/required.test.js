import { parse } from '@humanwhocodes/momoa';
import { describe, it, expect } from 'vitest';

import { getSchemaAndData } from '../../test-helpers';
import RequiredValidationError from '../required';

describe('Required', () => {
  it.each([
    ['prints correctly for missing required prop', true],
    ['prints correctly for missing required prop[without colors]', false],
  ])('%s', async (_, colorize) => {
    const [schema, data] = await getSchemaAndData('required', __dirname);
    const jsonRaw = JSON.stringify(data, null, 2);
    const jsonAst = parse(jsonRaw);

    const error = new RequiredValidationError(
      {
        keyword: 'required',
        dataPath: '/nested',
        schemaPath: '#/required',
        params: { missingProperty: 'id' },
        message: "should have required property 'id'",
      },
      { colorize, data, schema, jsonRaw, jsonAst },
    );

    expect(error.print()).toMatchSnapshot();
  });
});
