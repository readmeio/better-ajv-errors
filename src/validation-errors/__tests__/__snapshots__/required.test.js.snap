// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Required prints correctly for missing required prop 1`] = `
[
  "[31m[1mREQUIRED[22m[39m[31m should have required property 'id'[39m
",
  "  1 | {
> 2 |   "nested": {}
    |             ^ ☹️  [95mid[39m is missing here!
  3 | }",
]
`;

exports[`Required prints correctly for missing required prop[without colors] 1`] = `
[
  "REQUIRED should have required property 'id'
",
  "  1 | {
> 2 |   "nested": {}
    |             ^ ☹️  id is missing here!
  3 | }",
]
`;
